### PR TITLE
[FHL] Initial version of the SwiftUI Avatar Demo Controller.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This change introduces the SwiftUI Demo controller for the Avatar Vnext.
Avatar modifiers to set state properties were introduced.

### Verification

https://user-images.githubusercontent.com/68076145/127215305-d81b37bc-9fdd-4dd2-a057-5eaba18003f3.mov

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/656)

### CODE REVIEW

**Review by sophialee0416 — COMMENTED** [review]
This is really awesome!! Everything looks great. I was wondering if the Picker was able to stretch full width as well?

**Comment by rdeassis** [comment]
> This is really awesome!! Everything looks great. I was wondering if the Picker was able to stretch full width as well?

It seems like the Picker is very limited in iOS13. And despite I didn't spend a lot of time working on it, it seems like you can only clip it in case you want it shorter. But that still not an optimal solution.

On the iPhone it does take most of the width though:


Screen Shot 2021-07-27 at 2 32 59 PM

**Comment by Saadnajmi** [comment]
> > This is really awesome!! Everything looks great. I was wondering if the Picker was able to stretch full width as well?
> 
> It seems like the Picker is very limited in iOS13. And despite I didn't spend a lot of time working on it, it seems like you can only clip it in case you want it shorter. But that still not an optimal solution.
> 
> On the iPhone it does take most of the width though:
> 
> Screen Shot 2021-07-27 at 2 32 59 PM

The Picker is definitely a very.. nonextensible component. I solved a similar issue of needing to pick a bunch of items by making a button with a context menu. I think that'll only work for iOS 14+ though, and this menu might have too many items for that to be a good solution

**Comment by rdeassis** [comment]
Just pushed db57737e7f4995b6128b8c9ba0ae42d68d355a96 to address this issue that happens only on iOS14:


https://user-images.githubusercontent.com/68076145/127237599-dc05ab06-91f5-4dbb-afd3-523ed6630cc8.mov

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarDemoController_SwiftUI.swift [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                            .disableAutocorrection(true)
+                            .textFieldStyle(RoundedBorderTextFieldStyle())
+
+                        TextField("Secondary Text", text: $secondaryText)
+                            .autocapitalization(.none)
+                            .disableAutocorrection(true)
+                            .textFieldStyle(RoundedBorderTextFieldStyle())
+
+                        Toggle("Set image", isOn: $showImage)
+                            .toggleStyle(switchToggleStyle)
```
Comment (mischreiber) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarDemoController_SwiftUI.swift:91]: Nit: just to make this a bit cleaner (and while we don't yet have a vNext toggle control), would it be easier to define a custom view that automatically applies this property?

**Review by sophialee0416 — APPROVED** [review]

File: ios/FluentUI/Vnext/Avatar/Avatar.swift [hunk 2]
```diff
@@ -86,6 +86,25 @@ public struct AvatarView: View {
...
+    /// Creates and initializes a SwiftUI Avatar
+    /// - Parameters:
+    ///   - style: The style of the avatar.
+    ///   - size: The size of the avatar.
+    ///   - image: The optional image that the avatar should display.
+    ///   - primaryText: The primary text used to calculate the Avatar initials.
+    ///   - secondaryText: The secondary text used to calculate the Avatar initials.
+    public init(style: MSFAvatarStyle,
+                size: MSFAvatarSize,
+                image: UIImage?,
```
Comment (mischreiber) [inline ios/FluentUI/Vnext/Avatar/Avatar.swift:98]: We can combine this with the other init by giving these three optionals default `nil` values.

File: ios/FluentUI/Vnext/Avatar/AvatarModifiers.swift [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        state.hasPointerInteraction = hasPointerInteraction
+        return self
+    }
+
+    func hasRingInnerGap(_ hasRingInnerGap: Bool) -> AvatarView {
+        state.hasRingInnerGap = hasRingInnerGap
+        return self
+    }
+
+    func image(_ image: UIImage?) -> AvatarView {
```
Comment (mischreiber) [inline ios/FluentUI/Vnext/Avatar/AvatarModifiers.swift:36]: I don't think we should need modifiers for arguments that can be set during `init` (so `style`, `size`, `image`, `primaryText`, and `secondaryText`)

Comment (rdeassis) [reply ios/FluentUI/Vnext/Avatar/AvatarModifiers.swift:36]: Yeah! I agree!

**Review by mischreiber — APPROVED** [review]

**Pull request merged**